pkg/inputs/mysqlbatch/input.go

```diff
@@ -406,33 +419,28 @@ func InitTablePosition(db *sql.DB, positionCache position_store.PositionCacheInt
 // Pick primary key, if there is only one primary key
 // If pk not found try using unique index
 // fail
-func DetectScanColumn(sourceDB *sql.DB, dbName string, tableName string, maxFullDumpRowsCount int64) (string, int64, error) {
-	rowsCount, err := utils.EstimateRowsCount(sourceDB, dbName, tableName)
-	if err != nil {
-		return "", 0, errors.Trace(err)
-	}
-
+func DetectScanColumn(sourceDB *sql.DB, dbName string, tableName string, estimatedRowsCount int64, maxFullDumpRowsCount int64) (string, error) {
 	pks, err := utils.GetPrimaryKeys(sourceDB, dbName, tableName)
 	if err != nil {
-		return "", 0, errors.Trace(err)
+		return "", errors.Trace(err)
 	}
 
 	if len(pks) == 1 {
-		return pks[0], rowsCount, nil
+		return pks[0], nil
 	}
 
 	uniqueIndexes, err := utils.GetUniqueIndexesWithoutPks(sourceDB, dbName, tableName)
 	if err != nil {
-		return "", 0, errors.Trace(err)
+		return "", errors.Trace(err)
 	}
 
 	if len(uniqueIndexes) > 0 {
-		return uniqueIndexes[0], rowsCount, nil
+		return uniqueIndexes[0], nil
 	}
 
-	if rowsCount < maxFullDumpRowsCount {
-		return "*", rowsCount, nil
+	if estimatedRowsCount < maxFullDumpRowsCount {
```

Comment:
should this be the first condition?

Reply:
no, added a comment in the code @Ryan-Git